lib/rspec/mocks/object_reference.rb

```diff
@@ -42,44 +42,67 @@ def self.anonymous_module?(mod)
     # Used when an object is passed to `object_double`.
     # Represents a reference to that object.
     #
-    # @private
+    # @param object
     class DirectObjectReference
       def initialize(object)
         @object = object
       end
 
+      # The objects desription (`#inspect`).
+      # @return String
       def description
         @object.inspect
       end
 
+      # Not relevant to a direct object reference.
       def const_to_replace
         raise ArgumentError,
               "Can not perform constant replacement with an object."
       end
 
+      # The target of the verifying double (the object's class).
+      #
+      # @return Class
       def target
         @object.class
       end
 
+      # Always returns true for an object as the class is defined.
+      #
+      # @return true
       def defined?
         true
       end
 
+      # Performs the callback straight away as the object is already
+      # loaded.
+      #
+      # @yield object
       def when_loaded
         yield @object
       end
     end
 
-    # Used when an anonymous module is passed to `class_double` or `instance_double`.
+    # Used when a module is passed to `class_double` or `instance_double`.
```

Comment:
The "an anonymous" part is important, because that's the only case when this is used. When a named module is passed to `class_double` or `instance_double`, this is not used, but this comment now makes it sound like it would be.